core/kernel/tee_ta_manager.c

```diff
@@ -266,8 +266,13 @@ static void tee_ta_unlink_session(struct tee_ta_session *s,
 static void destroy_session(struct tee_ta_session *s,
 			    struct tee_ta_session_head *open_sessions)
 {
-	if (s->ctx)
+#if defined(CFG_TA_FTRACE_SUPPORT)
+	if (s->ctx) {
+		tee_ta_push_current_session(s);
 		ta_fbuf_dump(s);
+		s = tee_ta_pop_current_session();
```

Comment:
`s` must not change in this function. The only useful thing to do with return value of `tee_ta_pop_current_session()` is to check that it still matches `s`.

At some places in the code we assert that the returned value matches an expected value and at some places we just ignore it.

Reply:
ok, will just ignore it.